### ISSUE TITLE
Multiple experimenters request

### ISSUE THREAD

**Issue by lfrank** [NeurodataWithoutBorders/pynwb#985]
Feature request: 

Right now the "experimenter" field is a single string. It would be useful to have a standard for multiple experimenters, either a list of names separated by some delimiter or perhaps an array of strings.

**Comment by bendichter** [comment]
Yes, this would be helpful. This would allow datasets to be queried by experimenter more robustly and would help to ensure proper attribution of credit for data collection. I know the Chang Lab almost always has at least 2 experimenters per session, so this applies to other labs as well. The most straightforward thing to do would be to have a vector of strings, like we have for `keywords`, however I am worried that would not be backwards compatible if we simply changed the experimenter field to expect a vector of strings instead of a string. 

I think the ideal process for changing this would be
1) create an extension for this (maybe call it "experimenter**s**")
2) submit it as a community extension and have it go through the review process to be added to core
3) once experimenter**s** is in core deprecate experimenter in favor of experimenter**s** (even if just 1 experimenter is present and it's a vector of length 1). In the pynwb API we can continue to accept experimenter as an input and just construct experimenters on the fly so this won't break old conversion scripts.

As you mentioned @lfrank another option is to choose a delimiter, but this might cause issues down the road: the choice of delimiter might not be straightforward and might break in certain cases, this would be a little harder to query, and this would be an anti-pattern with the rest of the NWB standard.

If the extension plan sounds good I can create one.

**Comment by lfrank** [comment]
I'm happy with the extension idea, and if you're okay writing it, that would be great.

**Comment by bendichter** [comment]
EDIT: The ndx-experimenters isn't ready for use. It is causing a bug when using load_namespaces=True in NWBHDF5IO right now

I created this extension: https://github.com/bendichter/ndx-experimenters that allows you to do this:

```python
from ndx_experimenters import NWBFile
from datetime import datetime

nwb = NWBFile('session_description', 'identifier', datetime.now().astimezone(),
      experimenters=['a', 'b'])
```

I'm interested to hear others' thoughts on how I did this extension. Previously, (e.g. in [ndx-tasks](https://github.com/bendichter/ndx-task)) I added data to the `general` folder by extending `LabMetaData`. This allows you to add a `NWBGroup` (which may contain datasets), but does not allow you to add a `dataset` directly, which is what we want to do here. 

For this extenstion, I took a different approach, extending `NWBFile` to include an `experimenters` argument. The benefit of this is that `experimenters` in included in the file exactly where you would expect. It's a bit hacky though. In order to get the `NWBFile` neurodata_type to import properly, I had to actually name it something else in the YAML files (I used `NWBFile_experiments`). Another downside is that it would be tricky to do this type of extension multiple times for a single neurodata_type. For instance, if you want to add an `experimenters` argument and a similar `subject_ids` argument in the same manner, you would need to have the `subject_ids` extension import the `NWBFile` object from the `experimenters` extension. 

Maybe a better approach would be to change the core `NWBFile` to accept `NWBDataset` types
